discovery/pkg/sync/time.go

```diff
@@ -0,0 +1,22 @@
+package sync
+
+import "time"
+
+type nowFuncT func() time.Time
+
+var nowFunc nowFuncT
+
+func init() {
```

Comment:
Can we add some docs on why we need this and what it is doing?

Reply:
Just inline comments? We don't have any dev docs at the moment.

Reply:
Yep, sorry! That is what I meant :)

Reply:
I cleaned it up, removed unused methods, added the missing header, and added some comments.